apps/configutl.c

```diff
@@ -0,0 +1,200 @@
+/*
+ * Copyright 2025 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include <openssl/conf.h>
+#include <openssl/err.h>
+#include <openssl/safestack.h>
+
+#include "apps.h"
+#include "progs.h"
+
+/**
+ * Print the given value to stdout escaped for the OpenSSL configuration file
+ * format.
+ */
+static void print_escaped_value(BIO *out, const char *value)
+{
+    const char *p;
+
+    for (p = value; *p != 0; p++) {
+        switch (*p) {
+        case '"':
+        case '\'':
+        case '#':
+        case '\\':
+        case '$':
+            BIO_printf(out, "\\");
+            BIO_write(out, p, 1);
+            break;
+        case '\n':
+            BIO_printf(out, "%s", "\\n");
+            break;
+        case '\r':
+            BIO_printf(out, "%s", "\\r");
+            break;
+        case '\b':
+            BIO_printf(out, "%s", "\\b");
+            break;
+        case '\t':
+            BIO_printf(out, "%s", "\\t");
+            break;
+        case ' ':
+            if (p == value || p[1] == '\0') {
+                /*
+                 * Quote spaces if they are the first or last char of the
+                 * value. We could quote the entire string (and it would
+                 * certainly produce nicer output), but in quoted strings
+                 * the escape sequences for \n, \r, \t, and \b do not work.
+                 * To make sure we're producing correct results we'd thus
+                 * have to selectively not use those in quoted strings and
+                 * close and re-open the quotes if they appear, which is
+                 * more trouble than adding the quotes just around the
+                 * first and last leading and trailing space.
+                 */
+                BIO_printf(out, "%s", "\" \"");
```

Comment:
Might look a bit weird if there are multiple leading or trailing spaces

Reply:
It does look weird, but it preserves the value correctly.

Making it look nicer is certainly possible, but then requires special handling for the \n, \r, \t, and \b escape sequences, which according to my testing, don't work in double-quoted strings. So we'd either end up one of

```
directive = "  value including"\n"escape"\t"sequences  "
directive = " " value including\nescape\tsequences " "
directive = "  "value including\nescape\tsequences"  "
```

None of those strike me as particularly better than

```
directive = " " value including\nescape\tsequences " "
```

and values starting or ending with spaces are probably rare in practice.

Reply:
I'd keep it as is taking @neverpanic 's arguments reasonable